app/src/main/java/jembi/org/nfcdemo/utils/NdefFormatterTask.java

```diff
@@ -0,0 +1,78 @@
+package jembi.org.nfcdemo.utils;
+
+import android.nfc.FormatException;
+import android.nfc.NdefMessage;
+import android.nfc.NdefRecord;
+import android.nfc.Tag;
+import android.nfc.tech.Ndef;
+import android.os.AsyncTask;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+
+/**
+ * Created by Jembi Health Systems on 2017/10/09.
+ */
+
+public class NdefFormatterTask extends AsyncTask<Void, Void, Void> {
+
+    private Tag myTag;
+    private NfcFormatterCallback callback;
+
+    public NdefFormatterTask(Tag myTag, NfcFormatterCallback callback)  {
+        this.myTag = myTag;
+        this.callback = callback;
+    }
+
+    @Override
+    protected Void doInBackground(Void... voids) {
+        Ndef tag = Ndef.get(myTag);
+        try {
+            connectTag(tag);
+            tag.writeNdefMessage(createRecord());
+        } catch (FormatException | IOException ex) {
+            callback.onFormatError(ex);
+        } finally {
+            disconnectTag(tag);
+        }
+
+        return null;
+    }
+
+    @Override
+    protected void onPostExecute(Void aVoid) {
+        super.onPostExecute(aVoid);
+        callback.onFormatComplete();
+    }
+
+    private NdefMessage createRecord() throws UnsupportedEncodingException {
+
+        final String lang = "en";
+        byte[] textBytes = new byte[0];
+        byte[] langBytes = lang.getBytes("UTF-8");
+        int langLength = langBytes.length;
+        int textLength = textBytes.length;
+        final byte[] payload = new byte[1 + langLength + textLength];
+        return new NdefMessage(new NdefRecord(NdefRecord.TNF_WELL_KNOWN, NdefRecord.RTD_TEXT, new byte[0], payload));
+    }
+
+    private void connectTag(Ndef tag) throws IOException {
```

Comment:
This method handles IOException, so shouldn't be throwing it also. However, I'd recommend it'd be cleaner (i.e. less code) to just throw IOException and let the caller (doInBackground method) manage the exception because there is no special exception handling logic